test/lib/secretFactory.test.js

```diff
@@ -0,0 +1,232 @@
+            datastore.get.withArgs({
+                table: 'secrets',
+                params: {
+                    id
+                }
+            }).yieldsAsync(null, secretData);
+            hashaMock.sha1.returns(id);
+            ironMock.unseal.yieldsAsync(null, unsealed);
+        });
+
+        it('calls datastore get with id and returns correct values', () =>
+            factory.get(id)
+                .then(model => {
+                    assert.calledWith(ironMock.unseal, sealed, password, 'defaults');
+                    assert.instanceOf(model, Secret);
+                    assert.isTrue(datastore.get.calledOnce);
+                    Object.keys(expected).forEach(key => {
+                        assert.strictEqual(model[key], expected[key]);
+                    });
+                })
+        );
+
+        it('calls datastore get with config.id and returns correct values', () =>
+            factory.get({ id })
+                .then(model => {
+                    assert.calledWith(ironMock.unseal, sealed, password, 'defaults');
+                    assert.instanceOf(model, Secret);
+                    assert.isTrue(datastore.get.calledOnce);
+                    Object.keys(expected).forEach(key => {
+                        assert.strictEqual(model[key], expected[key]);
+                    });
+                })
+        );
+
+        it('calls datastore get with id generated from config and returns correct values', () =>
+            factory.get({ pipelineId, name })
+                .then(model => {
+                    assert.calledWith(ironMock.unseal, sealed, password, 'defaults');
+                    assert.instanceOf(model, Secret);
+                    assert.isTrue(datastore.get.calledOnce);
+                    Object.keys(expected).forEach(key => {
+                        assert.strictEqual(model[key], expected[key]);
+                    });
+                })
+        );
+    });
+
+    describe('list', () => {
+        const paginate = {
+            page: 1,
+            count: 2
+        };
+        const secret1 = {
+            id: '151c9b11e4a9a27e9e374daca6e59df37d8cf00f',
+            pipelineId: 'pipeline1',
+            name: 'secret1',
+            value: 'sealedsecret1value',
+            allowInPR: true
+        };
+        const secret2 = {
+            id: 'd398fb192747c9a0124e9e5b4e6e8e841cf8c71c',
+            pipelineId: 'pipeline1',
+            name: 'secret2',
+            value: 'sealedsecret2value',
+            allowInPR: false
+        };
+        const datastoreReturnValue = [secret1, secret2];
+        const returnValue = [secret1, secret2];
+
+        returnValue[0].value = 'batman';
+        returnValue[1].value = 'superman';
+
+        it('calls datastore scan and returns correct values', () => {
+            datastore.scan.yieldsAsync(null, datastoreReturnValue);
+            ironMock.unseal.withArgs(secret1.value, password).yieldsAsync(null, 'batman');
```

Comment:
This may not make a big difference, since it is just the mock data and mock execution, but I'm not convinced by this code that the correct values are being passed to unseal. Please change `secret1.value` to `sealedsecret1value`. Likewise with with `secret2.value`.